lib/composables/Url.ts

```diff
@@ -94,3 +94,29 @@ function unflattenObject(obj: Record<string, any>) {
     return acc
   }, {} as Record<string, any>)
 }
+
+function deepAssign(target: Record<string, any>, source: Record<string, any>) {
+  const dest = target
+  const src = toRaw(source)
+
+  if (isPlainObject(src)) {
+    Object.keys(src).forEach((key) => {
+      if (typeof src[key] === 'object' && src[key] !== null) {
+        deepAssign(dest[key], src[key])
+      } else {
+        dest[key] = toRaw(src[key])
+      }
+    })
+  } else if (Array.isArray(src)) {
+    dest.length = src.length
+    src.forEach((item, index) => {
+      if (typeof item === 'object' && item !== null) {
+        deepAssign(dest[index], item)
+      } else {
+        dest[index] = toRaw(item)
+      }
+    })
+  } else {
+    throw new TypeError('[deepAssign] src must be an object or array')
```

Comment:
Haven't implemented the other types yet. We can implement them as they come.

Reply:
Sounds good 👍